dxp/liferay_global_event_handler.md

```diff
@@ -0,0 +1,213 @@
+
+Description from: [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L476)
+
+Removes all listeners from the specified event. If the event type is not specified, all listeners from all hosted custom events will be removed.
+
+### `Liferay.publish`
+
+Description from: [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L559)
+
+Creates a new custom event of the specified type. If a custom event by that name already exists, it will not be re-created. In either case the custom event is returned.
+
+##### Code Example:
+
+```js
+Liferay.publish('allPortletsReady', {
+	fireOnce: true
+});
+
+// On another file located somewhere on DXP
+
+Liferay.on('allPortletsReady', function (event) {
+	document.getElementById('<%= HtmlUtil.escape(scroll) %>').scrollIntoView();
+});
+```
+
+### `Liferay.fire`
+
+Description from: [yui3/event-target.js at master · liferay/yui3 · GitHub](https://github.com/liferay/yui3/blob/master/src/event-custom/js/event-target.js#L723)
+
+Fire a custom event by name. The callback functions will be executed
+from the context specified when the event was created, and with the
+following parameters.
+
+The first argument is the event type, and any additional arguments are
+passed to the listeners as parameters. If the first of these is an
+object literal, and the event is configured to emit an event facade,
+that object is mixed into the event facade and the facade is provided
+in place of the original object.
+
+If the custom event object hasn't been created, then the event hasn't
+been published and it has no subscribers. For performance sake, we
+immediate exit in this case. This means the event won't bubble, so
+if the intention is that a bubble target be notified, the event must
+be published on this object first.
+
+##### Code example:
+
+```js
+Liferay.fire('showNavigationMenu', mapHover);
+```
+
+## Best Practices
+
+The best practice number #1 is using Liferay’s namespace utilities(Liferay.Util.ns) when possible. Like `Liferay.Util.ns(myWidgetNamespace, 'nameOfMyCustomEvent')`. For scoping the global name of the event. It will prevent you for listening to generic events which can be fired in different widgets that you may not be responsible.
+
+_Additional information_: When using simpler es6 class components, we could just extend to [PortletBase](https://github.com/liferay/liferay-portal/blob/master/modules/apps/frontend-js/frontend-js-web/src/main/resources/META-INF/resources/liferay/PortletBase.es.js) and use `this.ns` function.
+
+Wrong:
+
+```js
+Liferay.fire('CLOSE_MODAL', {
+	id: 'myId'
+});
+```
+
+Correct:
+
+```js
+import {ns} from 'frontend-js-web'; // or simply Liferay.Util.ns
+
+Liferay.fire(ns(portletNamespace,'CLOSE_MODAL'), {
+	id: 'myId'
+})
+
+// when not portletNamespace available
+
+Liferay.fire('commerce:CLOSE_MODAL'), {
```

Comment:
looks like you have an extra `)` here, which would probably cause a syntax error

Reply:
Thanks <3